### DISCUSSION TITLE
Creature Addon

### DISCUSSION THREAD

**Discussion by ghost** [TrinityCore/TrinityCore#25906] [General]
Regarding Addon tables, I'm assuming it would be best to have the addon data in the creature_template_addon table if the creature only has one spawn?

Would you want them left in the creature_addon table if they have paths?

Looking for some direction as I am automating a cleanup process.
Also regarding format below should aura values be cleaned up and standardized to "" or NULL if empty?

```
-- Wolfguard Worg Entry: 5058
DELETE FROM `creature_addon` WHERE `guid`= 16241;
DELETE FROM `creature_template_addon` WHERE `entry`= 5058;
INSERT INTO `creature_template_addon` (`entry`,`path_id`,`mount`,`bytes1`,`bytes2`,`emote`,`visibilityDistanceType`,`auras`) VALUES
(5058, 0, 0, 0, 4097, 0, 0, NULL); or (5058, 0, 0, 0, 4097, 0, 0, '');

**Comment by offl** [comment]
https://github.com/TrinityCore/TrinityCore/issues/25903#issuecomment-764862793
'' looks always better than NULL imo

**Reply by ghost** [reply]
Offi said:

Here's the story: bytes2 (sheath state) was always 0 as default, I changed very recently it to 1 as default
Someone added visibilityDistanceType and created thousands of c_t_a with sheathstate 0 (no weapon), updated visibilityDistanceType in c_a and leaved both c_a and c_t_a
So if you are going to update them all I'm not sure if this will help us or will make it worse
But in general it's right direction to convert single-used c_a to c_t_a

Maybe you need to skip all if c_t_a exists

**Reply by ghost** [reply]
So my conversion process is to check for all entries missing a c_t_a, Then check to see if that entry has only one spawn and if so check to see if that spawn has a c_a. If it does I convert the c_a to a c_t_a. I can also do a fix for '' to NULL for auras if you wish or just leave it as is.

**Reply by offl** [reply]
Yeah, sounds good
> I can also do a fix for '' to NULL for auras if you wish or just leave it as is.

Leave them as ''

You also need to decide when exactly we will start to apply those fixes to both branches because master does not have classic guids
Or it will be only 335?

**Reply by ghost** [reply]
I can do a fix for master separately. I just need to install the db and the software will do the work..

**Reply by offl** [reply]
Then 335 only
You can ask someone to push to master because I don't even have master compiled
